### PR TITLE
Handle firestore data changes

### PR DESCRIPTION
**Pull request by elayabharath**

### CODE REVIEW

**Comment by lukechurch** [comment]
@marianamarasoiu can I ask you to do a first review of this one?

File: lib/controller.dart [hunk 1]
```diff
@@ -342,9 +428,13 @@ void _computeFilterDropdownsAndRender(List<model.Filter> filters) {
           // todo: check across all _messageStats, not just the first
           var dates = _messageStats.values.first.keys.toList()..sort();
           filterOptions = [dates.first, dates.last];
-          defaultValue = dates.first.toString().split('T').first +
+          var splitCharacter = 'T';
+          if (!dates.first.toString().contains('T')) {
```
Comment (marianamarasoiu) [inline lib/controller.dart]: nit:
```suggestion
          if (!dates.first.toString().contains(splitCharacter)) {
```

File: lib/controller.dart [hunk 2]
```diff
@@ -117,20 +118,102 @@ void init() async {
...
+      _allInteractions = interactionsMap;
+      _handleDataChanges();
+    }, (error) {
+      view.showAlert(UNABLE_TO_FETCH_INTERACTIONS_ERROR_MSG);
+      logger.error(error.toString());
+    });
+  }
+}
+
+void _handleSurveyStatusChanges() {
```
Comment (marianamarasoiu) [inline lib/controller.dart:153]: these look very similar with each other - I think they fall under the issue of handling the different data types together, so let's keep these as they are for now and generalise next?

**Review by marianamarasoiu — APPROVED** [review]
thanks @elayabharath !

File: lib/controller.dart (hunk 2 above)
Comment (elayabharath) [reply lib/controller.dart:153]: Yep will generalise it along with storing the data collections in the controller

**Comment by elayabharath** [comment]
thanks @marianamarasoiu

**Pull request merged**